agent_build/package_builders.py

```diff
@@ -343,15 +343,24 @@ def _install_info_str(self) -> str:
 
     @staticmethod
     def _add_config(
-        config_source_path: Union[str, pl.Path], output_path: Union[str, pl.Path]
+        base_config_source_path: Union[str, pl.Path],
+        output_path: Union[str, pl.Path],
+        additional_config_paths: List[pl.Path] = None,
     ):
         """
         Copy config folder from the specified path to the target path.
+        :param base_config_source_path: Path to the base config directory to copy.
+        :param additional_config_paths: List of paths to config directories to copy to the result config directory.
+            New files are places along with files from the 'base_config_source_path' or overwrite them.
         """
-        config_source_path = pl.Path(config_source_path)
+        base_config_source_path = pl.Path(base_config_source_path)
         output_path = pl.Path(output_path)
         # Copy config
-        shutil.copytree(config_source_path, output_path)
+        shutil.copytree(base_config_source_path, output_path)
+
+        # Copy additional config files.
+        for a_config_path in additional_config_paths or []:
+            shutil.copytree(a_config_path, output_path, dirs_exist_ok=True)
 
         # Make sure config file has 640 permissions
         config_file_path = output_path / "agent.json"
```

Comment:
We should probably also do the same for other config files (set permissions to 640 for all config fragments in base_config_source_path), but that's not a blocker for this PR and just an improvement over the existing behavior.

Reply:
Those files also have to be checked, because that checking code goes right after :)

https://github.com/scalyr/scalyr-agent-2/blob/54e52da290fa263adf8fd6a032ccb1ea01779837/agent_build/package_builders.py#L361-L379